File: coresuite-dotnet-core-console-vb/generator/PackagePdf.vb
Imports System.IO
Imports ceTe.DynamicPDF
Imports ceTe.DynamicPDF.PageElements

Public Class PackagePdf

    Public Shared Sub Run(outputPdfPath As String)
        ' Create a PDF Document
        Dim MyDocument As Document = New Document
        MyDocument.Creator = "PackagePdf"
        MyDocument.Author = "ceTe Software"
        MyDocument.Title = "Package Pdf"

        ' FileStreams used for creating the instance of EmbeddedFile
        Dim fileStream1 As FileStream = New FileStream(Util.GetResourcePath("Images/DPDFLogo.png"), FileMode.Open, FileAccess.Read, FileShare.ReadWrite)
        Dim fileStream2 As FileStream = New FileStream(Util.GetResourcePath("Data/HelloWorldExcel.xls"), FileMode.Open, FileAccess.Read, FileShare.ReadWrite)

        ' byte array of the file by reading the file.
        Dim filebyte(fileStream2.Length) As Byte
        fileStream2.Read(filebyte, 0, filebyte.Length)

        ' Created 3 instances of EmbeddedFile using all the available constructors.
        Dim embeddedFile1 As EmbeddedFile = New EmbeddedFile(Util.GetResourcePath("PDFs/DocumentA.pdf"))
        Dim embeddedFile2 As EmbeddedFile = New EmbeddedFile(fileStream1, "DPDFLogo.png", DateTime.Now)
        Dim embeddedFile3 As EmbeddedFile = New EmbeddedFile(filebyte, "HelloWorldExcel.xls", DateTime.Now)

        ' Added the embeddedFiles to the EmbeddedFileList of the document class.
        MyDocument.EmbeddedFiles.Add(embeddedFile1)
        MyDocument.EmbeddedFiles.Add(embeddedFile2)
        MyDocument.EmbeddedFiles.Add(embeddedFile3)

        ' Create a Page and add it to the document
        Dim MyPage As Page = New Page
        MyDocument.Pages.Add(MyPage)

        ' Create a Document Package with Attachment Layout Detailed
        MyDocument.Package = New DocumentPackage(AttachmentLayout.Detailed)

        ' Add a label to the page
        MyPage.Elements.Add(New Label("This is the cover page for a package PDF", 0, 0, 512, 40, Font.Helvetica, 24, TextAlign.Center))

        ' Save the PDF document
        MyDocument.Draw(outputPdfPath)

    End Sub

End Class

File: coresuite-dotnet-framework-console-cs/generator/PackagePdf.cs
using ceTe.DynamicPDF;
using ceTe.DynamicPDF.PageElements;
using System;
using System.IO;

namespace coresuite_dotnet_framework_console_cs.Generator
{
    public class PackagePdf
    {
        public static void Run(string outputPdfPath)
        {
            // Create a PDF Document
            Document document = new Document
            {
                Creator = "PackagePdf",
                Author = "ceTe Software",
                Title = "Package Pdf"
            };

            // FileStreams used for creating the instance of EmbeddedFile
            FileStream fileStream1 = new FileStream(Util.GetResourcePath("Images/DPDFLogo.png"), FileMode.Open, FileAccess.Read, FileShare.ReadWrite);
            FileStream fileStream2 = new FileStream(Util.GetResourcePath("Data/HelloWorldExcel.xls"), FileMode.Open, FileAccess.Read, FileShare.ReadWrite);

            // byte array of the file by reading the file.
            byte[] filebyte = new byte[fileStream2.Length];
            fileStream2.Read(filebyte, 0, filebyte.Length);

            // Created 3 instances of EmbeddedFile using all the available constructors.
            EmbeddedFile embeddedFile1 = new EmbeddedFile(Util.GetResourcePath("PDFs/DocumentA.pdf"));
            EmbeddedFile embeddedFile2 = new EmbeddedFile(fileStream1, "DPDFLogo.png", DateTime.Now);
            EmbeddedFile embeddedFile3 = new EmbeddedFile(filebyte, "HelloWorldExcel.xls", DateTime.Now);

            // Added the embeddedFiles to the EmbeddedFileList of the document public class.
            document.EmbeddedFiles.Add(embeddedFile1);
            document.EmbeddedFiles.Add(embeddedFile2);
            document.EmbeddedFiles.Add(embeddedFile3);

            // Create a Page and add it to the document
            Page page = new Page();
            document.Pages.Add(page);

            // Create a Document Package with Attachment Layout Detailed
            document.Package = new DocumentPackage(AttachmentLayout.Detailed);

            // Add a label to the page 
            page.Elements.Add(new Label("This is the cover page for a package PDF", 0, 0, 512, 40, Font.Helvetica, 24, TextAlign.Center));

            // Save the PDF document
            document.Draw(outputPdfPath);
        }
    }
}
